infra/base-images/base-runner/test_all.py

```diff
@@ -0,0 +1,176 @@
+  Subprocess.ProcessResult."""
+  print('INFO: performing bad build checks for', fuzz_target)
+  command = ['bad_build_check', fuzz_target]
+  return subprocess.run(command,
+                        stderr=subprocess.PIPE,
+                        stdout=subprocess.PIPE,
+                        check=False)
+
+
+def get_broken_fuzz_targets(bad_build_results, fuzz_targets):
+  """Returns a list of broken fuzz targets and their process results in
+  |fuzz_targets| where each item in |bad_build_results| is the result of
+  bad_build_check on the corresponding element in |fuzz_targets|."""
+  broken = []
+  for result, fuzz_target in zip(bad_build_results, fuzz_targets):
+    if result.returncode != 0:
+      broken.append((fuzz_target, result))
+  return broken
+
+
+def has_ignored_targets(out_dir):
+  """Returns True if |out_dir| has any fuzz targets we are supposed to ignore
+  bad build checks of."""
+  out_files = set(os.listdir(out_dir))
+  ignored_targets = {
+      'do_stuff_fuzzer'
+      'checksum_fuzzer', 'fuzz_dump', 'fuzz_keyring', 'xmltest',
+      'fuzz_compression_sas_rle'
+  }
+  if out_files.intersection(ignored_targets):
+    return True
+  for filename in out_files:
+    if re.match(r'ares_*_fuzzer', filename):
+      return True
+  return False
+
+
+def test_all(pool, out, initial_out):
+  """Do bad_build_check on all fuzz targets."""
+  # TODO(metzman): Refactor so that we can conver test_one to python.
+  # Use a fake OUT directory to catch path hardcoding that breaks on
+  # ClusterFuzz.
+  recreate_directory(TMP_FUZZER_DIR)
+  try:
+    # Set this so that run_fuzzer which is called by bad_build_check works
+    # properly.
+    move_directory_contents(initial_out, out)
+    fuzz_targets = find_fuzz_targets(out)
+    bad_build_results = pool.map(do_bad_build_check, fuzz_targets)
+  finally:
+    move_directory_contents(out, initial_out)
+    out = initial_out
+
+  broken_targets = get_broken_fuzz_targets(bad_build_results, fuzz_targets)
+  broken_targets_count = len(broken_targets)
+  if not broken_targets_count:
+    return True
+
+  print('Broken fuzz targets ', broken_targets_count)
+  total_targets_count = len(fuzz_targets)
+  broken_targets_percentage = 100 * broken_targets_count / total_targets_count
+  for broken_target, result in broken_targets:
+    print(broken_target)
+    sys.stdout.buffer.write(result.stdout + result.stderr + b'\n')
+
+  allowed_broken_targets_percentage = int(
+      os.getenv('ALLOWED_BROKEN_TARGETS_PERCENTAGE', '10'))
+  if broken_targets_percentage > allowed_broken_targets_percentage:
+    print('ERROR: {broken_targets_percentage}% of fuzz targets seem to be '
+          'broken. See the list above for a detailed information.'.format(
+              broken_targets_percentage=broken_targets_percentage))
+    if has_ignored_targets(out):
+      return True
+    return False
+  print('{total_targets_count} fuzzers total, {broken_targets_count} '
```

Comment:
Use f'' style strings

Reply:
Can't. base-runner has python3.5 installed. That doesn't support fstrings.
Probably should hoist install base-image or install in base-runner as well.